### ISSUE TITLE
Fails to replace parent-selector when it's last and preceded by a universal pseudo selector

### ISSUE THREAD

**Issue by philpl** [thysultan/stylis.js#86]
This is an oddly specific bug that I found while playing around with the nested selector concatenation.

```js
const css = stylis('.test', `
  color: red;

  :hover & {
    color: green;
  }
`) // .test{color:red;}.test:hover &{color:green;}
```

This example results in `.test:hover &` for the inner rule.

```js
const css = stylis('.test', `
  color: red;

  div:hover & {
    color: green;
  }
`) // .test{color:red;}div:hover .test{color:green;}
```

However, when the universal pseudo selector is preceded by something else it returns the expected result. In this case: `div:hover .test`

**Comment by thysultan** [comment]
Thanks for finding this. Fixed in 3.4.7

**Comment by philpl** [comment]
Mate, you’re way too fast! 😂 Cheers!
